qai_hub_models/utils/args.py

```diff
@@ -1006,3 +1007,45 @@ def enable_model_caching(parser):
         " If overwrite, ignores and overwrites previous cache with newly uploaded AI Hub model instead.",
     )
     return parser
+
+
+def create_model_identifier(default_model_name: str, model_cls: type[FromPretrainedTypeVar],
+                                   model_kwargs: dict[str, Any]=None, *, custom_identifier_args: Optional[Tuple[str] | str] = None) -> str:
+    """
+    Parse the name of the weights provided by the user if any,
+    and appends the default model name to it,
+    otherwise return just the given `default_model_name`.
+
+    Args:
+        custom_identifier_args (List[str], optional): List of arguments to be used to identify the model.
+        default_model_name (str): The default name to use if no weights are provided.
+        model_cls (type[FromPretrainedTypeVar]): The model class with a `from_pretrained` method.
+        **model_kwargs: Keyword arguments passed in by the user.
+
+    Returns:
+        str: A name derived from the provided weights, or the default model name.
+    """
+
+    if not hasattr(model_cls, "from_pretrained") or model_kwargs is None:
+        return default_model_name
+
+    from_pretrained_sig = inspect.signature(model_cls.from_pretrained)
+
+    identifier_args =  (
+        custom_identifier_args
```

Comment:
One of the challenges with this change is tracking what these args are. They can vary significantly for each model.